students/laud/session04/kata_fourteen.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+import sys
+import random
+import string
+
+# Strip out punctuations, etc
+def cleanup(text):
+    lowercase = text.lower()
+    exclude = set(string.punctuation)
+    out = ''.join( ch for ch in lowercase if ch not in exclude)
+    return out
+
+# Read txt file
+def read_in_data(filename):
+    with open(filename, 'r') as text:
+        data = text.read()
+        clean = cleanup(data)
+    return clean
+
+# Split text into list
+def make_words(in_data):
+    data = in_data.split()
+    return data
+
+# Build key-value pairs
+def build_trigram(words):
+    trigrams = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+        if pair not in trigrams:
+           trigrams[pair] = []
+        trigrams[pair].append(follower)
+    return trigrams
+
+# Generate new text
+def build_text(word_pairs):
+    word_list = []
+    starting_pair = list(random.choice(list(word_pairs)))
+    word_list.extend(starting_pair)
+    for x in range(len(word_list)):
```

Comment:
Does this `for` loop work? At the time of the loop's beginning, `len(word_list)` is fixed to 2, and I guess this loop will just iterate twice, not any more? Basically your `word_list` is modified inside the loop, and you are using its length as the range in the loop header, which is very undesirable, if not incorrect. I think you might want to fix the number of iterations of this `for` loop somehow, like just hard-coding a fixed number or using a passed parameter?

Reply:
@hosungs Thanks. I'll take another look.

Reply:
Yes, I'm out of town the whole week this week, but I'll be teaching the class on Wednesday through Zoom, so we can handle this on Wednesday during the class time.

Reply:
@hosungs Sounds good. Thank you.